### DISCUSSION TITLE
token.nbor() IndexError (return None instead?)

### DISCUSSION THREAD

**Discussion by rfugal** [explosion/spaCy#7815] [New Features & Project Ideas]
Would it make sense to return None instead of IndexError when calling token.nbor() on the last token of a document?

I'm running at list comprehension which failed with IndexError
`for word, pos, ws, nbor in [(tkn.text, tkn.pos_, tkn.whitespace_, tkn.nbor()) for tkn in doc]:`

Now I need to do a workaround.

**Comment by polm** [comment]
If `Token.nbor` just returned the next token that might make sense, but it also takes an argument so it can access arbitrary words in the Doc. Since that's the case having it work the same way a list does seems like the most reasonable thing to do. 

If you don't want to reformat your code drastically the easiest thing for you here might be to make a little helper function called `get_next` or something that works around this.

**Reply by rfugal** [reply]
No problem, makes sense; the workaround is easy. 
`for tkn in doc:`
&emsp;`word, pos, ws = tkn.text, tkn.pos_, tkn.whitespace_`
&emsp;`try:`
&emsp;&emsp;`nbor = tkn.nbor()`
&emsp;`except IndexError:`
&emsp;&emsp;`nbor = None`
